dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/ClusterUtils.java

```diff
@@ -84,7 +84,9 @@ public static URL mergeUrl(URL remoteUrl, Map<String, String> localMap) {
             String remoteGroup = map.get(Constants.GROUP_KEY);
             String remoteRelease = map.get(Constants.RELEASE_KEY);
             map.putAll(localMap);
-            map.put(Constants.GROUP_KEY, remoteGroup);
+            if (!map.containsKey(Constants.GROUP_KEY) || StringUtils.isNotEmpty(remoteGroup)) {

```

Comment:
I think we should only check `StringUtils.isNotEmpty(remoteGroup)` but not `!map.containsKey(Constants.GROUP_KEY)`